src/video/canvas/texture.js

```diff
@@ -140,12 +140,22 @@
                     // Source coordinates
                     var s = frame.frame;
 
+                    var originX, originY;
+                    // Pixel-based offset origin from the top-left of the source frame
+                    var hasTextureAnchorPoint = (frame.spriteSourceSize && frame.sourceSize && frame.pivot);
+                    if (hasTextureAnchorPoint) {
+                        originX = (frame.sourceSize.w * frame.pivot.x) - frame.spriteSourceSize.x;
+                        originY = (frame.sourceSize.h * frame.pivot.y) - frame.spriteSourceSize.y;
+                    }
+                    
                     atlas[frame.filename] = {
-                        name    : name, // frame name
-                        offset  : new me.Vector2d(s.x, s.y),
-                        width   : s.w,
-                        height  : s.h,
-                        angle   : (frame.rotated === true) ? nhPI : 0
+                        name         : name, // frame name
+                        offset       : new me.Vector2d(s.x, s.y),
+                        hasTextureAnchorPoint : hasTextureAnchorPoint,
+                        anchorPoint  : (hasTextureAnchorPoint) ? new me.Vector2d(originX / s.w, originY / s.h) : null,
```

Comment:
OH! I see. I'm not crazy. The AnimationSheet inherits an anchorPoint from renderable. The boolean is there so we know whether the anchorPoint came from the texture, or whether it was the default anchorPoint. If it's from the texture, the sprite's position is drawn differently than it is today (reference entity.js)